ayon_server/settings/overrides.py

```diff
@@ -169,42 +169,66 @@ def extract_overrides(
     explicit_pins: list[list[str]] | None = None,
     explicit_unpins: list[list[str]] | None = None,
 ) -> dict[str, Any]:
+    existing_overrides = {**existing} if existing else {}
+
     result: dict[str, Any] = {}
 
```

Comment:
Looking at the logic and usage below it may be good to already here cache pins and unpins into a `set` for faster lookups.

```suggestion
    explicit_pins = set(explicit_pins) if explicit_pins else set()
    explicit_unpins = set(explicit_unpins) if explicit_unpins else set()
```

It would also make the code easier to read below because you don't need the `(explicit_pins or [])` logic there.

Reply:
explicit pins and unpins need to be list of lists as path is a list, which is unhashable. each path could be converted to set as well, but there are just too many places where path is list[str]....